--- Makefile
+++ Makefile
@@ -1,8 +1,8 @@
 all:
-	for i in libutf libfmt libbio libregexp mk; do (cd $$i; make); done; ls -l mk/mk
+	for i in libutf libfmt libbio libregexp mk; do (cd $$i; make PREFIX=$(PREFIX)); done; ls -l mk/mk
 
 clean:
-	for i in libutf libfmt libbio libregexp mk; do (cd $$i; make clean); done
+	for i in libutf libfmt libbio libregexp mk; do (cd $$i; make PREFIX=$(PREFIX) clean); done
 
 install:
-	for i in libutf libfmt libbio libregexp mk; do (cd $$i; make install); done
+	for i in libutf libfmt libbio libregexp mk; do (cd $$i; make PREFIX=$(PREFIX) install); done
