--- Makefile
+++ Makefile
@@ -1,11 +1,10 @@
 CC ?= cc
-CFLAGS ?= -O2
 override CFLAGS += -isystem . -D_GNU_SOURCE -DEMACS -DVI -std=gnu99 -Wall -Wno-format-security -Wno-pointer-sign
 LDFLAGS ?=
 DESTDIR ?= /
 PREFIX ?= /usr
 BIN_DIR ?= $(PREFIX)/bin
-BIN_NAME ?= ksh
+BIN_NAME ?= loksh
 MAN_DIR ?= $(PREFIX)/share/man
 DOC_DIR ?= $(PREFIX)/share/doc/loksh
 
@@ -19,23 +18,22 @@
 HEADERS = c_test.h charclass.h config.h edit.h expand.h ksh_limval.h lex.h \
           proto.h sh.h shf.h table.h tree.h tty.h
 
-all: ksh
+all: $(BIN_NAME)
 
 %.o: %.c $(HEADERS)
 	$(CC) -c -o $@ $< $(CFLAGS) $(NCURSES_CFLAGS)
 
-ksh: $(OBJECTS)
+$(BIN_NAME): $(OBJECTS)
 	$(CC) -o $@ $^ $(LDFLAGS) $(NCURSES_LDFLAGS)
 
 clean:
 	rm -f $(BIN_NAME) *.o
 
 install: all
-	install -v -D -m 755 ksh $(DESTDIR)/$(BIN_DIR)/$(BIN_NAME)
-	install -v -D -m 644 ksh.1 $(DESTDIR)/$(MAN_DIR)/man1/$(BIN_NAME).1
-	install -v -m 644 sh.1 $(DESTDIR)/$(MAN_DIR)/man1/sh.1
-	install -v -D -m 644 README $(DESTDIR)/$(DOC_DIR)/README
-	install -v -m 644 README.upstream $(DESTDIR)/$(DOC_DIR)/README.upstream
-	install -v -m 644 CONTRIBUTORS $(DESTDIR)/$(DOC_DIR)/CONTRIBUTORS
-	install -v -m 644 PROJECTS $(DESTDIR)/$(DOC_DIR)/PROJETCS
-	install -v -m 644 LEGAL $(DESTDIR)/$(DOC_DIR)/LEGAL
+	install -Dm0755 $(BIN_NAME) $(DESTDIR)/$(BIN_DIR)/$(BIN_NAME)
+	install -Dm0644 ksh.1 $(DESTDIR)/$(MAN_DIR)/man1/ksh.1
+	install -Dm0644 sh.1 $(DESTDIR)/$(MAN_DIR)/man1/sh.1
+	install -Dm0644  README $(DESTDIR)/$(DOC_DIR)/README
+	install -Dm0644 README.upstream $(DESTDIR)/$(DOC_DIR)/README.upstream
+	install -Dm0644 CONTRIBUTORS $(DESTDIR)/$(DOC_DIR)/CONTRIBUTORS
+	install -Dm0644 LEGAL $(DESTDIR)/$(DOC_DIR)/LEGAL
