--- src/utils.c	    2019-04-24 08:56:57.000000000 +0000
+++ src/utils.c.new 2019-07-13 14:15:07.403430254 +0000
@@ -250,7 +250,7 @@
 				regmatches[0].rm_so = next_rung;
 				regmatches[0].rm_eo = far_end;
 				if (regexec(&search_regexp, haystack, 1, regmatches,
-										REG_STARTEND) != 0)
+										0) != 0)
 					break;
 			}
 
@@ -258,7 +258,7 @@
 			regmatches[0].rm_so = floor;
 			regmatches[0].rm_eo = far_end;
 			if (regexec(&search_regexp, haystack, 10, regmatches,
-										REG_STARTEND) != 0)
+										0) != 0)
 				return NULL;
 
 			return haystack + regmatches[0].rm_so;
@@ -268,7 +268,7 @@
 		regmatches[0].rm_so = start - haystack;
 		regmatches[0].rm_eo = strlen(haystack);
 		if (regexec(&search_regexp, haystack, 10, regmatches,
-										REG_STARTEND) != 0)
+										0) != 0)
 			return NULL;
 		else
 			return haystack + regmatches[0].rm_so;
