--- pfetch	2020-04-04 13:58:31.362135374 -0500
+++ pfetch	2020-04-11 22:17:17.783901794 -0500
@@ -954,6 +954,21 @@
     #       allows indentation to continue naturally despite
     #       the use of '<<-EOF'.
     case ${1:-${PF_ASCII:-${distro:-$os}}} in
+        [Bb]onsai*)
+            read_ascii 4 <<-EOF
+${c2},####,
+${c2}#######,  ,#####,
+${c2}#####',#  '######
+${c2} ''###'${c3}';,,,${c2}'###'
+${c3}       ,;  ${c2}''''
+${c3}      ;;;   ${c2},#####,
+${c3}     ;;;'  ,,${c2};;;###
+${c3}     ';;;;'''${c2}####'
+${c3}      ;;;
+${c3}   ,.;;';'',,,
+${c3}  '     '
+			EOF
+        ;;
         [Aa]lpine*)
             read_ascii 4 <<-EOF
 				${c4}   /\\ /\\
