--- Makefile
+++ Makefile
@@ -11,11 +11,10 @@
 DIST=2bwm-$(VERSION)
 SRC=2bwm.c list.h hidden.c config.h
 DISTFILES=Makefile README.md TODO 2bwm.man $(SRC)
-CFLAGS+=-Os -s -I${X11_INCLUDE} \
-		-DTWOBWM_PATH=\"${TWOBWM_PATH}\" 
+CFLAGS+=-Os -s -I${X11_INCLUDE} -DTWOBWM_PATH=\"${TWOBWM_PATH}\" 
 
-LDFLAGS+=-L${PREFIX}/${LIB_SUFFIX} -lxcb -lxcb-randr -lxcb-keysyms \
-		 -lxcb-icccm -lxcb-ewmh -lxcb-xrm
+LDFLAGS+=-L${PREFIX}/${LIB_SUFFIX} -lxcb -lXau -lXdmcp -lxcb-randr -lxcb-keysyms -lxcb-icccm -lxcb-ewmh -lxcb-xrm -lxcb-util
+
 TARGETS=2bwm hidden
 OBJS=2bwm.o
 
