// thanks to dylan for this patch! <3
--- src/evdev.c
+++ src/evdev.c
@@ -40,7 +40,9 @@
 
 #include <linux/version.h>
 #include <sys/stat.h>
+#ifdef HAVE_UDEV
 #include <libudev.h>
+#endif
 #include <unistd.h>
 #include <errno.h>
 #include <fcntl.h>
@@ -226,6 +228,7 @@ EvdevIsDuplicate(InputInfoPtr pInfo)
 static BOOL
 EvdevDeviceIsVirtual(const char* devicenode)
 {
+#ifdef HAVE_UDEV
     struct udev *udev = NULL;
     struct udev_device *device = NULL;
     struct stat st;
@@ -256,6 +259,9 @@ out:
     udev_device_unref(device);
     udev_unref(udev);
     return rc;
+#else
+    return FALSE;
+#endif
 }
