--- Makefile    	2019-07-24 17:01:08.864080155 -0500
+++ Makefile_new	2019-07-24 17:14:18.470780056 -0500
@@ -2,63 +2,65 @@
 
 include config.mk
 
+# note: here we comment out all tools already
+# 		included between 9base and ubase
 SUBDIRS  = lib9\
 	yacc\
 	ascii\
 	awk\
-	basename\
 	bc\
 	cal\
-	cat\
-	cleanname\
-	cmp\
-	date\
 	dc\
-	du\
-	dd\
 	diff\
-	echo\
-	ed\
 	factor\
-	fortune\
 	fmt\
-	freq\
-	getflags\
-	grep\
-	hoc\
-	join\
-	listen1\
 	look\
-	ls\
-	md5sum\
-	mk\
-	mkdir\
-	mtime\
-	pbd\
-	primes\
-	rc\
-	read\
-	rm\
-	sam\
-	sha1sum\
-	sed\
-	seq\
-	sleep\
-	sort\
-	split\
-	ssam\
-	strings\
-	tail\
-	tee\
-	test\
-	touch\
-	tr\
-	troff\
-	unicode\
-	uniq\
-	unutf\
-	urlencode\
-	wc
+	mk
+#	basename\
+#	cat\
+#	cleanname\
+#	cmp\
+#	date\
+#	dd\
+#	du\
+#	echo\
+#	ed\
+#	fortune\
+#	freq\
+#	getflags\
+#	grep\
+#	hoc\
+#	join\
+#	listen1\
+#	ls\
+#	md5sum\
+#	mkdir\
+#	mtime\
+#	pbd\
+#	primes\
+#	rc\
+#	read\
+#	rm\
+#	sam\
+#	sha1sum\
+#	sed\
+#	seq\
+#	sleep\
+#	sort\
+#	split\
+#	ssam\
+#	strings\
+#	tail\
+#	tee\
+#	test\
+#	touch\
+#	tr\
+#	troff\
+#	unicode\
+#	uniq\
+#	unutf\
+#	urlencode\
+#	wc
 
 all:
 	@echo 9base build options:
