--- Makefile    	2019-07-07 19:46:27.516439190 +0000
+++ Makefile_new	2019-07-07 19:44:48.240310000 +0000
@@ -2,63 +2,65 @@
 
 include config.mk
 
-SUBDIRS  = lib9\
-	yacc\
-	ascii\
-	awk\
-	basename\
-	bc\
-	cal\
-	cat\
-	cleanname\
-	cmp\
-	date\
-	dc\
-	du\
-	dd\
-	diff\
-	echo\
-	ed\
-	factor\
-	fortune\
-	fmt\
-	freq\
-	getflags\
-	grep\
-	hoc\
-	join\
-	listen1\
-	look\
-	ls\
-	md5sum\
-	mk\
-	mkdir\
-	mtime\
-	pbd\
-	primes\
-	rc\
-	read\
-	rm\
-	sam\
-	sha1sum\
-	sed\
-	seq\
-	sleep\
-	sort\
-	split\
-	ssam\
-	strings\
-	tail\
-	tee\
-	test\
-	touch\
-	tr\
-	troff\
-	unicode\
-	uniq\
-	unutf\
-	urlencode\
-	wc
+SUBDIRS  = lib9 \
+	yacc \
+	ascii \
+	awk \
+	bc \
+	dc \
+	dd \
+	diff \
+	factor \
+	fmt \
+	look \
+	mk \
+	troff \
+	uniq
+
+	# rc \
+	# listen1\
+	# grep
+	# hoc\
+	# basename\
+	# cal\
+	# cat\
+	# cleanname\
+	# cmp\
+	# date\
+	# dc\
+	# du\
+	# echo\
+	# ed\
+	# fortune\
+	# freq\
+	# getflags\
+	# join\
+	# ls\
+	# md5sum\
+	# mkdir\
+	# mtime\
+	# pbd\
+	# primes\
+	# read\
+	# rm\
+	# sam\
+	# sha1sum\
+	# sed\
+	# seq\
+	# sleep\
+	# sort\
+	# split\
+	# ssam\
+	# strings\
+	# tail\
+	# tee\
+	# test\
+	# touch\
+	# tr\
+	# unicode\
+	# unutf\
+	# urlencode\
+	# wc
 
 all:
 	@echo 9base build options:
