--- Makefile
+++ Makefile
@@ -10,7 +10,7 @@
 CFLAGS?=-O2
 INCLUDES=-Iinclude
 
-SHAREDLIB=libnl-tiny.so
+# SHAREDLIB=libnl-tiny.so
 STATICLIB=libnl-tiny.a
 PCFILE=libnl-tiny.pc
 ALL_LIBS=$(SHAREDLIB) $(STATICLIB)
@@ -28,7 +28,7 @@
 all: $(ALL_LIBS) $(PCFILE)
 
 install: $(ALL_LIBS:%=$(DESTDIR)$(libdir)/%) \
-         $(ALL_INCLUDES:include/%=$(DESTDIR)$(includedir)/libnl-tiny/%) \
+         $(ALL_INCLUDES:include/%=$(DESTDIR)$(includedir)/%) \
          $(PCFILE:%=$(DESTDIR)$(libdir)/pkgconfig/%)
 
 clean:
@@ -38,8 +38,8 @@
 	$(CC) $(CPPFLAGS) -c -o $@ $(INCLUDES) $(CFLAGS) $(PICFLAGS) $<
 
 
-$(SHAREDLIB): $(OBJS)
-	$(CC) -shared -o $@ $^ $(LDFLAGS)
+# $(SHAREDLIB): $(OBJS)
+	# $(CC) -shared -o $@ $^ $(LDFLAGS)
 
 $(STATICLIB): $(OBJS)
 	rm -f $@
@@ -50,7 +50,7 @@
 	sed s,@prefix@,$(prefix),g $< > $@
 
 
-$(DESTDIR)$(includedir)/libnl-tiny/%: include/%
+$(DESTDIR)$(includedir)/%: include/%
 	$(INSTALL) -D -m 644 $< $@
 
 $(DESTDIR)$(libdir)/%: %
