--- crypto/compat/getprogname_linux.c	2019-05-07 07:05:51.000000000 +0000
+++ getprogname_linux.c	2019-07-02 23:57:06.827021642 +0000
@@ -26,9 +26,7 @@
 #if defined(__ANDROID_API__) && __ANDROID_API__ < 21
 	extern const char *__progname;
 	return __progname;
-#elif defined(__GLIBC__)
-	return program_invocation_short_name;
 #else
-#error "Cannot emulate getprogname"
+    return program_invocation_short_name;
 #endif
 }
