--- src/drivers/driver_nl80211.c	2014-08-22 21:38:40.456927876 +0300
+++ src/drivers/driver_nl80211.c	2014-08-22 21:39:16.583927229 +0300
@@ -60,6 +60,12 @@
 #define PACKET_TX_TIMESTAMP	16
 #endif
 
+#ifdef CONFIG_LIBNL_TINY
+#define nl_handle nl_sock
+#define nl_handle_alloc_cb nl_socket_alloc_cb
+#define nl_handle_destroy nl_socket_free
+#endif
+
 #ifdef ANDROID
 #include "android_drv.h"
 #endif /* ANDROID */
--- src/drivers/drivers.mak	2014-08-22 21:38:40.456927876 +0300
+++ src/drivers/drivers.mak	2014-08-22 21:39:16.583927229 +0300
@@ -39,6 +39,7 @@ ifdef CONFIG_LIBNL32
 else
   ifdef CONFIG_LIBNL_TINY
     DRV_LIBS += -lnl-tiny
+    DRV_CFLAGS += -DCONFIG_LIBNL_TINY
   else
     DRV_LIBS += -lnl
   endif
